lib/documents.js

```diff
@@ -269,7 +269,7 @@ function probeDocumentsImpl(contentOnly, args) {
   var connectionParams = this.client.getConnectionParams();
   var requestOptions = mlutil.copyProperties(connectionParams);
   requestOptions.method = 'HEAD';
-  requestOptions.path = mlutil.databaseParam(connectionParams, path, '&');
+  requestOptions.path = mlutil.newRequestOptions(connectionParams, path, '&');
```

Comment:
This seems unexpected - the function is called `newRequestOptions`, implying that it will return a "request options" object, but its return value is instead being assigned to the value of "path". 

I think what's shown in the JIRA ticket reads better and also removes more code:

```
const requestOptions = mlutil.newRequestOptions(
    connectionParams, path, '&');
```

That would replace line 270 above as well, and it would return a requestOptions object that has the path correctly constructed (which includes accounting for the optional "database" param). Each client of that function would then assign `method` and any other properties necessary.

Reply:
Note too that since `method` seems to always be assigned in this repeated block of code, it would be reasonable to add that as a param:

```
const requestOptions = mlutil.newRequestOptions(connectionParams, path, 'POST', '&');
```
```

Reply:
you mean we should include the copyProperties ?

Reply:
Okay so changing databaseParam -> newRequestOptions(connectionParams, endpoint, separator, method) that sets the method as well.

Reply:
Right, that's the goal of the refactor - try to condense a block of 4 or 5 lines that is frequently repeated into a single readable line of code.